src/Query/Terms.php

```diff
@@ -15,97 +15,90 @@
 class Terms extends AbstractQuery
 {
     /**
-     * Terms.
-     *
-     * @var array Terms
+     * @var string
      */
-    protected $_terms;
+    private $field;
 
     /**
-     * Terms key.
-     *
-     * @var string Terms key
+     * @var string[]
      */
-    protected $_key;
+    private $terms;
 
     /**
-     * Construct terms query.
-     *
-     * @param string $key   Terms key
-     * @param array  $terms Terms list
+     * @var string[]|null
```

Comment:
@thePanz @ruflin 

Is `string[]` really the only correct type?

Wouldn't be `string[]|int[]` be equally correct (and all the other references in this class)?

For ElasticSearch, AFAIK, it doesn't matter if I send a number as string or an actual int, both equally work well.

I'm asking because in the process of the upgrade, using static analyzers, now complains passing integers to these methods but from my experience technically this isn't strictly required?

Thank you

Reply:
I haven't tested it but I think you are correct @mfn That both will work. I assume even a double could be passed in and just works? Could you test it?

Reply:
So I did some testing with the `terms` query directly in the Kibana console and my findings are that string, int and float (i.e. the JavaScript equivalent) work equally well.

E.g. if I've field of type `long` and I use it in a simple query like this:
```json
GET …/_search
{
    "query": {
      "terms": {
        "some_field": [
          1234
        ]
      }
   }
}
```
Whether I use `"1234"`, `1234` or `1234.0` did not make any difference.

Note that the same applies to `term` query too, I tested this:
```json
GET …/_search
{
    "query": {
      "term": {
        "some_field": {
          "value":  1234
        }
      }
   }
}
```
(Ok realized the `Term` query class does not use these specific type hints, so not affected by this.)

---
I think since fields can have the type `double`, yes, I guess `float` is also appropriate to include (though not sure how useful a term search for this would be compared to e.g. ranges, but who knows).

Do you think this is good enough warranted a phpdoc change here, i.e. to `string[]|int[]|float[]`? Annoyingly long btw :)

Reply:
@mfn Yes! Can you open a PR?

Reply:
Yes!